sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java

```diff
@@ -1,26 +1,35 @@
 package com.kyriba.conference.sponsorship.api;
 
-import com.kyriba.conference.sponsorship.api.dto.PlanCancellationResponse;
-import com.kyriba.conference.sponsorship.api.dto.PlanRegistered;
-import com.kyriba.conference.sponsorship.api.dto.PlanRegistrationRequest;
-import com.kyriba.conference.sponsorship.domain.Plan;
+import com.kyriba.conference.sponsorship.domain.PlanCategory;
+import com.kyriba.conference.sponsorship.domain.dto.PlanDto;
 import com.kyriba.conference.sponsorship.service.PlanService;
 import io.swagger.annotations.Api;
 import io.swagger.annotations.ApiOperation;
 import io.swagger.annotations.ApiParam;
 import io.swagger.annotations.ApiResponse;
 import io.swagger.annotations.ApiResponses;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
 import lombok.RequiredArgsConstructor;
+import lombok.Value;
+import org.springframework.dao.EmptyResultDataAccessException;

```

Comment:
'dao' layer should not be used on 'controller/api' layer

Reply:
Good point. I moved it from a service to a controller, because  there should be "no HTTP knowledge at business logic level".  As a result, such a handler cannot sit in service, because it uses "HttpStatus" and cannot sit in a controller, because it handles dao exception. So I have a question then. Where and how should I use ExceptionHandler for dao exceptions? Is there any other option except custom exceptions?

Reply:
From service layer you should throw only custom business exceptions if you want other layer to rely on them. Else concrete technology changes and all external handlers become broken.